src/main/scala/com/campudus/tableaux/database/model/FileModel.scala

```diff
@@ -0,0 +1,200 @@
+          |filename,
+          |idfolder,
+          |created_at) VALUES (?,?,?,?,?,?,CURRENT_TIMESTAMP) RETURNING created_at""".stripMargin
+
+    for {
+      result <- connection.singleQuery(insert, Json.arr(uuid.toString, o.name, o.description, o.mimeType, o.filename, o.folder.orNull))
+      resultArr <- Future(insertNotNull(result).head)
+    } yield {
+      val createdAt = DateTime.parse(resultArr.get[String](0))
+      File(Some(uuid), o.name, o.description, o.mimeType, o.filename, o.folder, Some(createdAt), None)
+    }
+  }
+
+  override def retrieve(id: UUID): Future[File] = {
+    val select =
+      s"""SELECT
+          |uuid,
+          |name,
+          |description,
+          |mime_type,
+          |filename,
+          |idfolder,
+          |created_at,
+          |updated_at
+          |FROM $table WHERE
+          |uuid = ? AND tmp = FALSE""".stripMargin
+
+    for {
+      result <- connection.singleQuery(select, Json.arr(id.toString))
+      resultArr <- Future(selectNotNull(result))
+    } yield {
+      convertJsonArrayToFile(resultArr.head)
+    }
+  }
+
+  def convertJsonArrayToFile(arr: JsonArray): File = {
+    File(
+      arr.get[String](0), //uuid
+      arr.get[String](1), //name
+      arr.get[String](2), //description
+      arr.get[String](3), //mime_type
+      arr.get[String](4), //filename
+      arr.get[Long](5), //idfolder
+      arr.get[String](6), //created_at
+      arr.get[String](7) //updated_at
+    )
+  }
+
+  implicit def convertLongToFolderId(id: Long): Option[FolderId] = {
+    //TODO not cool!
+    if (id == 0) {
+      None
+    } else {
+      Some(id.toLong)
+    }
+  }
+
+  implicit def convertStringToUUID(str: String): Option[UUID] = {
+    Some(UUID.fromString(str))
+  }
+
+  implicit def convertStringToDateTime(str: String): Option[DateTime] = {
+    if (str == null) {
+      None
+    } else {
+      Option(DateTime.parse(str))
+    }
+  }
+
+  override def update(o: File): Future[File] = {
+    val update =
+      s"""UPDATE $table SET
+          |name = ?,
+          |description = ?,
+          |idfolder = ?,
+          |updated_at = CURRENT_TIMESTAMP,
+          |tmp = FALSE
+          |WHERE uuid = ? RETURNING mime_type, filename, created_at, updated_at""".stripMargin
+
+    for {
+      result <- connection.singleQuery(update, Json.arr(o.name, o.description, o.folder.orNull, o.uuid.get.toString))
+      resultArr <- Future.apply(updateNotNull(result))
+    } yield {
+      File(
+        o.uuid,
+        o.name,
+        o.description,
+        resultArr.head.get[String](0), //mime_type
+        resultArr.head.get[String](1), //filename
+        o.folder,
+        resultArr.head.get[String](2), //created_at
+        resultArr.head.get[String](3) //updated_at
+      )
+    }
+  }
+
+  override def size(): Future[Long] = {
+    val select = s"SELECT COUNT(*) FROM $table WHERE tmp = FALSE"
+
+    for {
+      result <- connection.singleQuery(select, Json.emptyArr())
```

Comment:
We should add a query method to `DatabaseConnection` that doesn't need a `JsonArray` for parameters. We could even send a regular query, not a prepared statement.